### PR TITLE
Updating usage of the --metrics flag to create a job

### PR DESCRIPTION
**Pull request by coreydaley**
Updated the --metrics flag of the oc cluster up command to create a job
so that it will automatically retry if it fails due to the service account
token not being created yet.

Fixes issue #11946

### CODE REVIEW

**Comment by coreydaley** [comment]
@csrwng @liggitt Please review

File: pkg/bootstrap/docker/openshift/metrics.go [hunk 1]
```diff
@@ -2,12 +2,12 @@ package openshift
...
 	"fmt"
-
+	"github.com/openshift/origin/pkg/bootstrap/docker/errors"
 	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
 	kapi "k8s.io/kubernetes/pkg/api"
 	apierrors "k8s.io/kubernetes/pkg/api/errors"
-
-	"github.com/openshift/origin/pkg/bootstrap/docker/errors"
+	kbatch "k8s.io/kubernetes/pkg/apis/batch"
+	"time"
```
Comment (csrwng) [inline pkg/bootstrap/docker/openshift/metrics.go]: Looks like left over from previous change

File: pkg/bootstrap/docker/openshift/metrics.go [hunk 2]
```diff
@@ -65,11 +65,35 @@ func (h *Helper) InstallMetrics(f *clientcmd.Factory, hostName, imagePrefix, ima
 		return errors.NewError("cannot create metrics deployer secret").WithCause(err).WithDetails(h.OriginLog())
 	}
 
-	// Create deployer Pod
+	// Create the job client
+	jobClient := kubeClient.Batch().Jobs(infraNamespace)
+
+	// Create the job
 	deployerPod := metricsDeployerPod(hostName, imagePrefix, imageVersion)
```
Comment (csrwng) [inline pkg/bootstrap/docker/openshift/metrics.go]: you should rename the function to metricsDeployerJob and return a Job instead of a Pod

**Comment by coreydaley** [comment]
@csrwng Code updated, thanks!

File: pkg/bootstrap/docker/openshift/metrics.go [hunk 3]
```diff
@@ -186,7 +187,27 @@ func metricsDeployerPod(hostName, imagePrefix, imageVersion string) *kapi.Pod {
...
 	}
-	return pod
+
+	completions := int32(1)
+
+	deadline := int64(60 * 5)
+
+	meta := kapi.ObjectMeta{
+		Name:      pod.Name,
+		Namespace: infraNamespace,
```
Comment (csrwng) [inline pkg/bootstrap/docker/openshift/metrics.go]: you don't need to set a namespace. The client will create it in the right namespace

File: pkg/bootstrap/docker/openshift/metrics.go [hunk 4]
```diff
@@ -186,7 +187,27 @@ func metricsDeployerPod(hostName, imagePrefix, imageVersion string) *kapi.Pod {
 			Env: env,
 		},
 	}
-	return pod
+
```
Comment (csrwng) [inline pkg/bootstrap/docker/openshift/metrics.go]: You really don't need a pod object anymore (https://github.com/openshift/origin/pull/12174/files#diff-8f476646dd3c6f6850aa06d1673053d7L147)

you should just create a pod spec and directly use it below

**Comment by coreydaley** [comment]
@csrwng updated

File: pkg/bootstrap/docker/openshift/metrics.go [hunk 5]
```diff
@@ -144,49 +145,65 @@ func metricsDeployerPod(hostName, imagePrefix, imageVersion string) *kapi.Pod {
...
-					Name:      "empty",
-					MountPath: "/etc/deploy",
-				},
+
+	completions := int32(1)
+
+	deadline := int64(60 * 5)
+
+	meta := kapi.ObjectMeta{
+		Name: metricsDeployerPodName,
```
Comment (csrwng) [inline pkg/bootstrap/docker/openshift/metrics.go]: rename constant to metricsDeployerJobName

File: pkg/bootstrap/docker/openshift/metrics.go [hunk 6]
```diff
@@ -2,12 +2,11 @@ package openshift
...
 import (
 	"fmt"
-
+	"github.com/openshift/origin/pkg/bootstrap/docker/errors"
 	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
 	kapi "k8s.io/kubernetes/pkg/api"
 	apierrors "k8s.io/kubernetes/pkg/api/errors"
-
-	"github.com/openshift/origin/pkg/bootstrap/docker/errors"
+	kbatch "k8s.io/kubernetes/pkg/apis/batch"
```
Comment (csrwng) [inline pkg/bootstrap/docker/openshift/metrics.go]: see if you can turn off the editor automatically rearranging the imports. 

We really should have something like:
```
	"fmt"

	kapi "k8s.io/kubernetes/pkg/api"
	apierrors "k8s.io/kubernetes/pkg/api/errors"
	kbatch "k8s.io/kubernetes/pkg/apis/batch"

	"github.com/openshift/origin/pkg/bootstrap/docker/errors"
	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
```

**Comment by coreydaley** [comment]
@csrwng sounds good to me, code has been updated.

**Comment by csrwng** [comment]
LGTM [merge]

**Comment by coreydaley** [comment]
test flake: #8571

**Comment by stevekuznetsov** [comment]
re[test]

**Comment by coreydaley** [comment]
[test]

**Comment by coreydaley** [comment]
flake #8502

**Comment by coreydaley** [comment]
flake #9203

**Comment by coreydaley** [comment]
[test]

**Comment by csrwng** [comment]
[merge]

**Comment by coreydaley** [comment]
flake https://github.com/openshift/origin/issues/12236

**Comment by coreydaley** [comment]
re[test]

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/12184

**Comment by bparees** [comment]
[test]

**Pull request merged**